torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -1234,6 +1353,30 @@ std::vector<CodeGen::CallArg> TensorExprKernel::prepareRunArgs(
     }
   }
 
+  if (has_symbolic_shapes_) {
+    // If there are symbolic shapes, then the output tensor size wouldn't have
+    // been computed at compile time. That has to be done here by using the
+    // symbolic shape input params passed in to this call.
+    TORCH_INTERNAL_ASSERT(
+        tensorOutputSymbolicSizes_.size() == bufOutputs_.size());
+    tensorOutputSizes_.resize(bufOutputs_.size());
```

Comment:
Why do we need to resize it on every invocation of the kernel? Shouldn't it have the appropriate size already?

Reply:
Only the first `resize` should have any effect. The subsequent ones should not do anything. 

I have now moved the `resize` to earlier, during compilation and made these asserts.